modules/apps/frontend-js/frontend-js-a11y-web/src/main/resources/META-INF/resources/Violations.tsx

```diff
@@ -0,0 +1,255 @@
+	{label: Liferay.Language.get('minor'), value: 'minor'},
+];
+
+const CATEGORY_FILTER_OPTIONS = [
+	{label: 'WCAG 2.0A', value: 'wcag2a'},
+	{label: 'WCAG 2.0 Level AA', value: 'wcag2aa'},
+	{label: 'WCAG 2.1 Level A', value: 'wcag21a'},
+	{label: 'WCAG 2.1 Level AA', value: 'wcag21aa'},
+	{label: 'Best Practices', value: 'best-practice'},
+];
+
+type ViolationsFilterProps = {
+	selectedCategories: Array<String>;
+	selectedImpact: Array<ImpactValue>;
+};
+
+function ViolationsFilter({
+	selectedCategories,
+	selectedImpact,
+}: ViolationsFilterProps) {
+	const [dropdownExpanded, setDropdownExpanded] = useState(false);
+	const dispatch = useFilteredViolationsDispatch();
+
+	return (
+		<>
+			<ClayDropDown
+				active={dropdownExpanded}
+				closeOnClickOutside
+				onActiveChange={setDropdownExpanded}
+				trigger={
+					<ClayButtonWithIcon
+						aria-label={Liferay.Language.get(
+							'open-violations-filter'
+						)}
+						displayType="unstyled"
+						small
+						symbol="filter"
+					/>
+				}
+			>
+				<ClayDropDown.ItemList>
+					<ClayDropDown.Group
+						header={Liferay.Language.get('filter-by-impact')}
+					>
+						{IMPACT_FILTER_OPTIONS.map((item) => {
+							const existsImpact =
+								selectedImpact.findIndex(
+									(element) => element === item.value
+								) !== -1;
+
+							return (
+								<ClayDropDown.Section
+									active={existsImpact}
+									key={item.value}
+								>
+									<ClayCheckbox
+										checked={existsImpact}
+										data-testid={item.value}
+										label={item.label}
+										onChange={() => {
+											if (existsImpact) {
+												dispatch({
+													payload: {
+														value: item.value,
+													},
+													type: TYPES.IMPACT_REMOVE,
+												});
+
+												return;
+											}
+
+											dispatch({
+												payload: {
+													value: item.value,
+												},
+												type: TYPES.IMPACT_ADD,
+											});
+										}}
+									></ClayCheckbox>
+								</ClayDropDown.Section>
+							);
+						})}
+					</ClayDropDown.Group>
+					<ClayDropDown.Group
+						header={Liferay.Language.get('filter-by-category')}
+					>
+						{CATEGORY_FILTER_OPTIONS.map((item) => {
+							const existsCategory =
+								selectedCategories.findIndex(
+									(element) => element === item.value
+								) !== -1;
+
+							return (
+								<ClayDropDown.Section key={item.value}>
+									<ClayCheckbox
+										checked={existsCategory}
+										data-testid={item.value}
+										label={item.label}
+										onChange={() => {
+											if (existsCategory) {
+												dispatch({
+													payload: {
+														value: item.value,
+													},
+													type: TYPES.CATEGORY_REMOVE,
+												});
+
+												return;
+											}
+											dispatch({
+												payload: {
+													value: item.value,
+												},
+												type: TYPES.CATEGORY_ADD,
+											});
```

Comment:
You can simplify this, since they have the same signature and avoid two `dispatch` statements.

```suggestion
											dispatch({
												payload: {
													value: item.value,
												},
												type: existsCategory ? TYPES.CATEGORY_REMOVE : TYPES.CATEGORY_ADD,
											});
```

Reply:
Good! It saves some LOC :)